### PR TITLE
Add Document get_index_from_location / get_location_from_index

### PR DESCRIPTION
**Pull request by davidbrochart**
**Please review the following checklist.**

- [x] Docstrings on all new or modified functions / classes 
- [ ] Updated documentation
- [x] Updated CHANGELOG.md (where appropriate)

### CODE REVIEW

**Comment by davidbrochart** [comment]
I think these methods could be useful when some document changes come from a system which is not `Location`-aware, but only reports changes pointing to raw indexes in the document content, like in collaborative editing.

**Comment by willmcgugan** [comment]
@darrenburns Could you have a look at this one pls.

File: src/textual/document/_document.py [hunk 1]
```diff
@@ -320,6 +342,40 @@ def line_count(self) -> int:
         """Returns the number of lines in the document."""
         return len(self._lines)
 
+    def get_index_from_location(self, location: Location) -> int:
```
Comment (darrenburns) [inline src/textual/document/_document.py:345]: I can see why both of these would be useful. Would you mind writing a few tests for them?

Also, we have a policy of avoiding shorthand variable names where possible, so if you wouldn't mind renaming `col` -> `column`, `i` -> `line_index`, etc. that'd be appreciated too.

**Review by darrenburns — CHANGES_REQUESTED** [review]

**Comment by davidbrochart** [comment]
Thanks for the review @darrenburns, I made the changes, rebased and added some tests.

**Comment by darrenburns** [comment]
Thanks @davidbrochart, could you also update the changelog?

**Comment by davidbrochart** [comment]
Done.

**Review by darrenburns — APPROVED** [review]

**Comment by darrenburns** [comment]
Thanks!

**Pull request merged**